pymc_marketing/mmm/base.py

```diff
@@ -30,36 +29,36 @@
 __all__ = ("BaseMMM", "MMM")
 
 
-class BaseMMM:
+class BaseMMM(ModelBuilder):
     model: pm.Model
 
     def __init__(
         self,
-        data: pd.DataFrame,
         target_column: str,
         date_column: str,
         channel_columns: Union[List[str], Tuple[str]],
+        data: Optional[pd.DataFrame] = None,
```

Comment:
can we leave `data` as a the first argument?

Reply:
Also, why can this be none? We should always have input data right?

Reply:
Not requiring the data as an argument makes the class more scikit-learn like, where the data needs to be available only for fit, also an idea I've discussed with Thomas is to utilise the generate_model_data, which this PR adds to allow sandbox-like experience, where the user can just initialise the class with minimal parameters to learn about the expected data structure and functionalities. But overall I'd say that making data optional is a step towards making the class more intuitive for scikit-learn users

Reply:
so in general, with the sandbox approach I'm basically implementing the generate_model_data in a way that it will return the example data used in the tests, in case no data would be provided. The users still keep the option of:
a) As actually the default example dataset should allow for not specifying the date_column and target_column, as in the default scenario it can be specified up front. In this case we don't have any non-default parameters and the data can be provided as a first argument, keeping the previous api unchanged. -> This requires creation of a wrapper function, read below
c)can provide the data directly to fit. In this case, the data needs to be passed in a X,y format to make it compatible with sklearn.
As I've realised now, If we still want to allow the users to pass the entire dataset as it's being done now, I'll need to add a wrapper function that will call fit under the hood while passing the data in a correct format, so far as the users would be forced to call fit by themselves, passing data to the class constructor makes very little sense, as they would anyway need to specify which columns of the dataset are the predictors. So in this case I need your opinion on how much we want to keep the old api. The easiest solution would be to just require users to pass the data in sklearn format to fit